client/api.c

```diff
@@ -704,6 +707,49 @@ TDNFOpenHandle(
                   TDNF_CONF_GROUP);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    if (!IsNullOrEmptyString(pTdnf->pArgs->pszInstallRoot) &&
+        strcmp(pTdnf->pArgs->pszInstallRoot, "/"))
+    {
+        int nIsDir = 0;
+
+        dwError = TDNFAllocateStringPrintf(&pszCacheDir, "%s/%s",
+                                           pTdnf->pArgs->pszInstallRoot,
+                                           pTdnf->pConf->pszCacheDir);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        TDNF_SAFE_FREE_MEMORY(pTdnf->pConf->pszCacheDir);
+        pTdnf->pConf->pszCacheDir = pszCacheDir;
+        pszCacheDir = NULL;
+
+        dwError = TDNFAllocateStringPrintf(&pszRepoDir, "%s/%s",
+                                           pTdnf->pArgs->pszInstallRoot,
+                                           pTdnf->pConf->pszRepoDir);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        dwError = TDNFIsDir(pszRepoDir, &nIsDir);
+        if (dwError == ERROR_TDNF_SYSTEM_BASE + ENOENT)
```

Comment:
Use ERROR_TDNF_FILE_NOT_FOUND from tdnferror.h.

Do the same in repoutils.c as well(in a different PR or piggyback it here, no issues). There are multiple entries of the same in repoutils.c